bot/twitter.py

```diff
@@ -67,7 +67,7 @@ def extract_and_save_tweets(
     elif user_id is None and username is None:
         raise TypeError("Expected ID or username or both, not none of them")
 
```

Comment:
I think it is not a good idea to generate a timestamp and reuse it throughout each function since that might reproduce the bug we already have!
(imagine we generate a timestamp then in one of the functions we face a rate limit after the rate limit we will use that timestamp, as this timestamp is not exactly 7 days ago and might return the error we already have)

It's a rare case but might happen